File: backend/app/models/items.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class ItemSummary(BaseModel):
    item_name: str
    total_games: int
    wins: int
    top4_count: int
    avg_placement: float
    most_common_champion: Optional[str] = None
    last_updated: Optional[datetime] = None


class ItemDetail(ItemSummary):
    pass


class ItemChampionCombo(BaseModel):
    champion_id: str
    total_games: int
    wins: int
    top4_count: int
    avg_placement: float
    win_rate: float


class ItemListResponse(BaseModel):
    items: List[ItemSummary]
    total: int

File: frontend/src/types/items.ts
export interface ItemSummary {
  item_name: string;
  total_games: number;
  wins: number;
  top4_count: number;
  avg_placement: number;
  most_common_champion: string | null;
  last_updated: string | null;
}

export interface ItemDetail extends ItemSummary {}

export interface ItemChampionCombo {
  champion_id: string;
  total_games: number;
  wins: number;
  top4_count: number;
  avg_placement: number;
  win_rate: number;
}

export interface ItemListResponse {
  items: ItemSummary[];
  total: number;
}
